### PR TITLE
Fix rendering of the Castle Captain's spell points bar

### LINKED ISSUE #2984: Kingdom Overview screen, Towns/Castles screen, for the Captain's portrait sprite, the indicator bar for the number of Captain's spell points is too big compared to what it should be
In fheroes2, for the Captain's portrait sprite, the indicator bar for the number of Captain's spell points is too big compared to what it should be.

In original game:

Spell points bar captain H2 01

In fheroes2:

Spell points bar captain fh2 01

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
Fix #2984

### CODE REVIEW

File: src/fheroes2/castle/captain.cpp [hunk 1]
```diff
@@ -277,6 +279,12 @@ void Captain::PortraitRedraw( s32 px, s32 py, PortraitType type, fheroes2::Image
 
     // spell points
     fheroes2::Fill( dstsf, px + barWidth + port.width() + 2, py, barWidth, iconHeight, blueColor );
-    const fheroes2::Sprite & mana = fheroes2::AGG::GetICN( ICN::MANA, GetMaxSpellPoints() );
     fheroes2::Blit( mana, dstsf, px + barWidth + port.width() + 2, py + mana.y() );
 }
+
+int Captain::GetManaIndexSprite( void ) const
```
Comment (ihhub) [inline src/fheroes2/castle/captain.cpp:285]: Please do not use `void` as an input parameter.

File: src/fheroes2/castle/captain.cpp [hunk 2]
```diff
@@ -277,6 +279,12 @@ void Captain::PortraitRedraw( s32 px, s32 py, PortraitType type, fheroes2::Image
...
+{
+    // valid range (0 - 25)
+    int r = GetMaxSpellPoints() / 5;
```
Comment (ihhub) [inline src/fheroes2/castle/captain.cpp:288]: We could make this variable `const`.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @oleg-derevenetz , just 2 comments left. Could you please take a look?

File: src/fheroes2/castle/captain.cpp (hunk 1 above)
Comment (oleg-derevenetz) [reply src/fheroes2/castle/captain.cpp:285]: Oh, I've seen it at many places, and I thought that it's a project style. No problem, I'll fix this.

**Comment by ihhub** [comment]
@oleg-derevenetz , thank you so much for fixing this bug!

**Pull request merged**